operator/builtin/transformer/recombine/recombine.go

```diff
@@ -123,12 +128,31 @@ type RecombineOperator struct {
 	overwriteWithOldest bool
 	combineField        entry.Field
 	combineWith         string
+	ticker              *time.Ticker
+	timeout             time.Duration
+	chClose             chan struct{}
 
 	sync.Mutex
 	batch []*entry.Entry
 }
 
 func (r *RecombineOperator) Start(_ operator.Persister) error {
+	go func() {
+		for {
+			select {
+			case <-r.ticker.C:
+				r.Lock()
+				if err := r.flushCombined(); err != nil {
+					r.Errorw("Failed flushing", "error", err)
+				}
+				r.Unlock()
+			case <-r.chClose:
+				r.ticker.Stop()
+				return
+			}
+		}
+	}()
+
```

Comment:
Can I ask you to flatten this somewhat, perhaps by pulling out a function?

Reply:
@djaglowski I've pulled that out to a `func`. Let me know if it's better now.